File: backend/src/main/java/model/Nota.java
package model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Nota {

	@Id
	@GeneratedValue(strategy=GenerationType.AUTO)
	private Long id;
	private float valor;
	private int peso;
	private String descricao;
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public float getValor() {
		return valor;
	}
	public void setValor(float valor) {
		this.valor = valor;
	}
	public int getPeso() {
		return peso;
	}
	public void setPeso(int peso) {
		this.peso = peso;
	}
	public String getDescricao() {
		return descricao;
	}
	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}
	
	
	/*
	private Materia materia;
	spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
	*/
}

File: frontend/src/app/models/nota.ts
import { Materia } from './materia';

export class Nota {
  id: number;
  valor: number;
  peso: number;
  descricao: string;
  materia: Materia;

  constructor(id: number, nome: string, valor: number, peso: number, descricao: string, materia: Materia) {
    this.id = id;
    this.valor = valor;
    this.peso = peso;
    this.descricao = descricao;
    this.materia = materia;
  }
}
